packages/kit/src/runtime/client/client.js

```diff
@@ -1078,14 +1069,34 @@ async function load_root_error_page({ status, error, url, route }) {
 		data: null
 	};
 
-	return await get_navigation_result_from_branch({
-		url,
-		params,
-		branch: [root_layout, root_error],
-		status,
-		error,
-		route: null
-	});
+	try {
+		const root_layout = await load_node({
+			loader: default_layout_loader,
+			url,
+			params,
+			route,
+			parent: () => Promise.resolve({}),
+			server_data_node: create_data_node(server_data_node)
+		});
+
+		return await get_navigation_result_from_branch({
+			url,
+			params,
+			branch: [root_layout, root_error],
+			status,
+			error,
+			route: null
+		});
+	} catch (err) {
+		return await get_navigation_result_from_branch({
+			url,
+			params,
+			branch: [root_error],
+			status,
+			error,
+			route: null
+		});
+	}
```

Comment:
```suggestion
	} catch (err) {
		return await get_navigation_result_from_branch({
			url,
			params,
			branch: [root_error],
			status,
			error,
			route: null
		});
	}
```
I don't think this is quite right. We need to differentiate between Redirect errors and other errors instead of loading the default error page without the root layout results. A redirect should navigate to the redirected URL while other errors should render the static fallback error page.